azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,429 @@
+
+   private static int runningNumber = 1;
+   private final String azureGroup;
+
+   @Resource
+   @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+   private Logger logger = Logger.NULL;
+
+   private final AzureComputeApi api;
+
+   private final AzureComputeConstants azureComputeConstants;
+
+   @Inject
+   AzureComputeServiceAdapter(final AzureComputeApi api, final AzureComputeConstants azureComputeConstants,
+                              @Named("azureGroupId") String group) {
+
+      this.api = api;
+      this.azureComputeConstants = azureComputeConstants;
+      this.azureGroup = group;
+   }
+
+   @Override
+   public NodeAndInitialCredentials<VMDeployment> createNodeWithGroupEncodedIntoName(
+           final String group, final String name, final Template template) {
+
+      final String location = template.getLocation().getId();
+      DeploymentTemplateBuilder deploymentTemplateBuilder = api.deploymentTemplateFactory().create(group, name, template);
+
+      final String loginUser = deploymentTemplateBuilder.getLoginUserUsername();
+      final String loginPassword = deploymentTemplateBuilder.getLoginPassword();
+
+      DeploymentBody deploymentTemplateBody =  deploymentTemplateBuilder.getDeploymentTemplate();
+
+      DeploymentProperties properties = DeploymentProperties.create(deploymentTemplateBody);
+
+      final String deploymentTemplate = UrlEscapers.urlFormParameterEscaper().escape(deploymentTemplateBuilder.getDeploymentTemplateJson(properties));
+
+      logger.debug("Deployment created with name: %s", name);
+
+      final Set<VMDeployment> deployments = Sets.newHashSet();
+
+      ResourceGroupApi resourceGroupApi = api.getResourceGroupApi();
+      ResourceGroup resourceGroup = resourceGroupApi.get(azureGroup);
+      final String resourceGroupName;
+
+      if (resourceGroup == null){
+         final Map<String, String> tags = ImmutableMap.of("description", "jClouds managed VMs");
+         resourceGroupName = resourceGroupApi.create(azureGroup, location, tags).name();
+      } else {
+         resourceGroupName = resourceGroup.name();
+      }
+
+      if (!retry(new Predicate<String>() {
+         @Override
+         public boolean apply(final String name) {
+            runningNumber++;
+
+            Deployment deployment = api.getDeploymentApi(resourceGroupName).create(name, deploymentTemplate);
+
+            if (deployment != null) {
+               VMDeployment vmDeployment = new VMDeployment();
+               vmDeployment.deployment = deployment;
+               deployments.add(vmDeployment);
+            } else {
+               logger.debug("Failed to create deployment!");
+            }
+            return !deployments.isEmpty();
+         }
+      }, azureComputeConstants.operationTimeout(), 1, SECONDS).apply(name)) {
+         final String illegalStateExceptionMessage = format("Deployment %s was not created within %sms so it will be destroyed.",
```

Comment:
Thanks @ritazh! 
I think you can extract the `destroyNode` logic into a function like [CleanupServer](https://github.com/jclouds/jclouds/blob/a4a255fa4ae6c7a1a46aa2a8420b19cd2b7df463/apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/functions/CleanupServer.java)
which is actually a `CleanupAccount` :)

This can be used in the `destroyNode` and in `AzureComputeService` similarly to [this](https://github.com/jclouds/jclouds/blob/a4a255fa4ae6c7a1a46aa2a8420b19cd2b7df463/apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/NovaComputeService.java#L99-L99)

makes sense?